diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57f9c7c..a88ed26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,7 +72,6 @@ fcitx5_add_i18n_definition()
 add_subdirectory(modules)
 add_subdirectory(im)
 add_subdirectory(po)
-add_subdirectory(tools)
 
 if (ENABLE_GUI)
     add_subdirectory(gui)
diff --git a/im/pinyin/pinyin.cpp b/im/pinyin/pinyin.cpp
index 6dd8f2e..2119ff4 100644
--- a/im/pinyin/pinyin.cpp
+++ b/im/pinyin/pinyin.cpp
@@ -688,8 +688,7 @@ std::string PinyinEngine::evaluateCustomPhrase(InputContext *inputContext,
 
 PinyinEngine::PinyinEngine(Instance *instance)
     : instance_(instance),
-      factory_([this](InputContext &) { return new PinyinState(this); }),
-      worker_(instance->eventDispatcher()) {
+      factory_([this](InputContext &) { return new PinyinState(this); }) {
     ime_ = std::make_unique<libime::PinyinIME>(
         std::make_unique<libime::PinyinDictionary>(),
         std::make_unique<libime::UserLanguageModel>(
@@ -847,25 +846,18 @@ void PinyinEngine::loadDict(const std::string &fullPath,
     }
     ime_->dict()->addEmptyDict();
     PINYIN_DEBUG() << "Loading pinyin dict " << fullPath;
-    std::packaged_task<libime::PinyinDictionary::TrieType()> task([fullPath]() {
-        std::ifstream in(fullPath, std::ios::in | std::ios::binary);
-        auto trie = libime::PinyinDictionary::load(
-            in, libime::PinyinDictFormat::Binary);
-        return trie;
-    });
-    taskTokens.push_back(worker_.addTask(
-        std::move(task),
-        [this, index = ime_->dict()->dictSize() - 1, fullPath](
-            std::shared_future<libime::PinyinDictionary::TrieType> &future) {
-            try {
-                PINYIN_DEBUG()
-                    << "Load pinyin dict " << fullPath << " finished.";
-                ime_->dict()->setTrie(index, future.get());
-            } catch (const std::exception &e) {
-                PINYIN_ERROR() << "Failed to load pinyin dict " << fullPath
-                               << ": " << e.what();
-            }
-        }));
+    std::ifstream in(fullPath, std::ios::in | std::ios::binary);
+    auto trie = libime::PinyinDictionary::load(
+        in, libime::PinyinDictFormat::Binary);
+    auto index = ime_->dict()->dictSize() - 1;
+    try {
+        PINYIN_DEBUG()
+            << "Load pinyin dict " << fullPath << " finished.";
+        ime_->dict()->setTrie(index, trie);
+    } catch (const std::exception &e) {
+        PINYIN_ERROR() << "Failed to load pinyin dict " << fullPath
+                        << ": " << e.what();
+    }
 }
 
 void PinyinEngine::loadBuiltInDict() {
diff --git a/im/pinyin/pinyin.h b/im/pinyin/pinyin.h
index 6dcf9bb..57ec6df 100644
--- a/im/pinyin/pinyin.h
+++ b/im/pinyin/pinyin.h
@@ -498,7 +498,6 @@ private:
     std::unique_ptr<HandlerTableEntry<EventHandler>> event_;
     CustomPhraseDict customPhrase_;
     SymbolDict symbols_;
-    WorkerThread worker_;
     std::list<std::unique_ptr<TaskToken>> persistentTask_;
     std::list<std::unique_ptr<TaskToken>> tasks_;
 
