diff --git a/im/pinyin/pinyin.cpp b/im/pinyin/pinyin.cpp
index 513a8d4..343870a 100644
--- a/im/pinyin/pinyin.cpp
+++ b/im/pinyin/pinyin.cpp
@@ -754,8 +754,7 @@ std::string PinyinEngine::evaluateCustomPhrase(InputContext *inputContext,
 
 PinyinEngine::PinyinEngine(Instance *instance)
     : instance_(instance),
-      factory_([this](InputContext &) { return new PinyinState(this); }),
-      worker_(instance->eventDispatcher()) {
+      factory_([this](InputContext &) { return new PinyinState(this); }) {
     ime_ = std::make_unique<libime::PinyinIME>(
         std::make_unique<libime::PinyinDictionary>(),
         std::make_unique<libime::UserLanguageModel>(
@@ -913,6 +912,7 @@ void PinyinEngine::loadDict(const std::string &fullPath,
     }
     ime_->dict()->addEmptyDict();
     PINYIN_DEBUG() << "Loading pinyin dict " << fullPath;
+#if 0
     std::packaged_task<libime::PinyinDictionary::TrieType()> task([fullPath]() {
         std::ifstream in(fullPath, std::ios::in | std::ios::binary);
         auto trie = libime::PinyinDictionary::load(
@@ -932,6 +932,19 @@ void PinyinEngine::loadDict(const std::string &fullPath,
                                << ": " << e.what();
             }
         }));
+#endif
+    std::ifstream in(fullPath, std::ios::in | std::ios::binary);
+    auto trie = libime::PinyinDictionary::load(
+        in, libime::PinyinDictFormat::Binary);
+    auto index = ime_->dict()->dictSize() - 1;
+    try {
+        PINYIN_DEBUG()
+            << "Load pinyin dict " << fullPath << " finished.";
+        ime_->dict()->setTrie(index, trie);
+    } catch (const std::exception &e) {
+        PINYIN_ERROR() << "Failed to load pinyin dict " << fullPath
+                        << ": " << e.what();
+    }
 }
 
 void PinyinEngine::loadBuiltInDict() {
diff --git a/im/pinyin/pinyin.h b/im/pinyin/pinyin.h
index 5ace603..0ee8798 100644
--- a/im/pinyin/pinyin.h
+++ b/im/pinyin/pinyin.h
@@ -518,7 +518,6 @@ private:
     std::unique_ptr<HandlerTableEntry<EventHandler>> event_;
     CustomPhraseDict customPhrase_;
     SymbolDict symbols_;
-    WorkerThread worker_;
     std::list<std::unique_ptr<TaskToken>> persistentTask_;
     std::list<std::unique_ptr<TaskToken>> tasks_;
 
diff --git a/modules/cloudpinyin/fetch.cpp b/modules/cloudpinyin/fetch.cpp
index a634736..af5c246 100644
--- a/modules/cloudpinyin/fetch.cpp
+++ b/modules/cloudpinyin/fetch.cpp
@@ -245,5 +245,4 @@ void FetchThread::run() {
     // free events ahead of time
     timer_.reset();
     events_.clear();
-    loop_.reset();
 }
diff --git a/tools/scel2org5.cpp b/tools/scel2org5.cpp
index 2bcdba7..bf9e5a7 100644
--- a/tools/scel2org5.cpp
+++ b/tools/scel2org5.cpp
@@ -497,7 +497,7 @@ void readDelTable(std::istream &in, const ScelOption &options) {
 
 } // namespace
 
-int main(int argc, char **argv) {
+extern "C" __attribute__((used)) int scel2org5(int argc, char *argv[]) {
     int c;
     std::optional<std::string_view> outputFile;
     bool printDel = false;
