diff --git a/im/pinyin/pinyin.cpp b/im/pinyin/pinyin.cpp
index 0592124..eab9f20 100644
--- a/im/pinyin/pinyin.cpp
+++ b/im/pinyin/pinyin.cpp
@@ -710,8 +710,7 @@ std::string PinyinEngine::evaluateCustomPhrase(InputContext *inputContext,
 
 PinyinEngine::PinyinEngine(Instance *instance)
     : instance_(instance),
-      factory_([this](InputContext &) { return new PinyinState(this); }),
-      worker_(instance->eventDispatcher()) {
+      factory_([this](InputContext &) { return new PinyinState(this); }) {
     ime_ = std::make_unique<libime::PinyinIME>(
         std::make_unique<libime::PinyinDictionary>(),
         std::make_unique<libime::UserLanguageModel>(
@@ -869,6 +868,7 @@ void PinyinEngine::loadDict(const std::string &fullPath,
     }
     ime_->dict()->addEmptyDict();
     PINYIN_DEBUG() << "Loading pinyin dict " << fullPath;
+#if 0
     std::packaged_task<libime::PinyinDictionary::TrieType()> task([fullPath]() {
         std::ifstream in(fullPath, std::ios::in | std::ios::binary);
         auto trie = libime::PinyinDictionary::load(
@@ -888,6 +888,19 @@ void PinyinEngine::loadDict(const std::string &fullPath,
                                << ": " << e.what();
             }
         }));
+#endif
+    std::ifstream in(fullPath, std::ios::in | std::ios::binary);
+    auto trie = libime::PinyinDictionary::load(
+        in, libime::PinyinDictFormat::Binary);
+    auto index = ime_->dict()->dictSize() - 1;
+    try {
+        PINYIN_DEBUG()
+            << "Load pinyin dict " << fullPath << " finished.";
+        ime_->dict()->setTrie(index, trie);
+    } catch (const std::exception &e) {
+        PINYIN_ERROR() << "Failed to load pinyin dict " << fullPath
+                        << ": " << e.what();
+    }
 }
 
 void PinyinEngine::loadBuiltInDict() {
diff --git a/im/pinyin/pinyin.h b/im/pinyin/pinyin.h
index a190794..750ab80 100644
--- a/im/pinyin/pinyin.h
+++ b/im/pinyin/pinyin.h
@@ -516,7 +516,6 @@ private:
     std::unique_ptr<HandlerTableEntry<EventHandler>> event_;
     CustomPhraseDict customPhrase_;
     SymbolDict symbols_;
-    WorkerThread worker_;
     std::list<std::unique_ptr<TaskToken>> persistentTask_;
     std::list<std::unique_ptr<TaskToken>> tasks_;
 
diff --git a/modules/cloudpinyin/fetch.cpp b/modules/cloudpinyin/fetch.cpp
index 4c1dff3..7749de8 100644
--- a/modules/cloudpinyin/fetch.cpp
+++ b/modules/cloudpinyin/fetch.cpp
@@ -247,5 +247,4 @@ void FetchThread::run() {
     // free events ahead of time
     timer_.reset();
     events_.clear();
-    loop_.reset();
 }
