diff --git a/CMakeLists.txt b/CMakeLists.txt
index a9d9c7e..03782c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,13 +10,12 @@ ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 message ("CMAKE_INSTALL_PREFIX = ${CMAKE_INSTALL_PREFIX}")
 
 find_package(ECM 1.0.0 REQUIRED)
-set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
+set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
 option(ENABLE_QT "Enable Qt for GUI configuration" On)
 option(USE_QT6 "Build against Qt6" On)
 include(FeatureSummary)
 include(GNUInstallDirs)
 include(ECMUninstallTarget)
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 # See Fcitx5UtilsConfig.cmake
 add_definitions(-DFCITX_INSTALL_USE_FCITX_SYS_PATHS=ON)
