diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 6c380a7..d3f2972 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -24,6 +24,7 @@ target_link_libraries(libime_tabledict LibIME::Table)
 install(TARGETS libime_tabledict DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT tools)
 add_executable(LibIME::tabledict ALIAS libime_tabledict)
 
+if (0)
 add_executable(libime_migrate_fcitx4_table libime_migrate_fcitx4_table.cpp)
 target_link_libraries(libime_migrate_fcitx4_table LibIME::Table)
 install(TARGETS libime_migrate_fcitx4_table DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT tools)
@@ -33,3 +34,4 @@ add_executable(libime_migrate_fcitx4_pinyin libime_migrate_fcitx4_pinyin.cpp)
 target_link_libraries(libime_migrate_fcitx4_pinyin LibIME::Pinyin)
 install(TARGETS libime_migrate_fcitx4_pinyin DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT tools)
 add_executable(LibIME::migrate_fcitx4_pinyin ALIAS libime_migrate_fcitx4_pinyin)
+endif()
