diff --git a/src/rimeengine.cpp b/src/rimeengine.cpp
index cdaaa3d..dcddf74 100644
--- a/src/rimeengine.cpp
+++ b/src/rimeengine.cpp
@@ -48,6 +48,16 @@
 #include <utility>
 #include <vector>
 
+namespace fcitx {
+enum class Runtime {
+    web = 0,
+    webworker = 1,
+    serviceworker = 2, // ChromeOS
+};
+extern Runtime runtime;
+extern int deployRimeInWorker();
+}
+
 FCITX_DEFINE_LOG_CATEGORY(rime_log, "rime");
 
 namespace fcitx::rime {
@@ -287,13 +297,26 @@ void RimeEngine::rimeStart(bool fullcheck) {
 
     fcitx_rime_traits.modules = nullptr;
 
+    bool autoDeploy = false;
     if (firstRun_) {
         api_->setup(&fcitx_rime_traits);
         firstRun_ = false;
+        if (runtime == Runtime::web) {
+            autoDeploy = !fs::isreg(userDir / "build/default.yaml");
+        }
     }
     api_->initialize(&fcitx_rime_traits);
     api_->set_notification_handler(&rimeNotificationHandler, this);
-    api_->start_maintenance(fullcheck);
+    /*          web      web worker  service worker
+    start  default.yaml?     no            no
+    reload       no          no            no
+    deploy   !useWorker     yes            no
+    */
+    if (fullcheck || autoDeploy) {
+        if (!deployRimeInWorker()) {
+            api_->start_maintenance(fullcheck);
+        }
+    }
 
     if (!api_->is_maintenance_mode()) {
         updateAppOptions();
@@ -521,12 +544,13 @@ void RimeEngine::notify(RimeSessionId session, const std::string &messageType,
     bool blockMessage = false;
     if (messageType == "deploy") {
         tipId = "fcitx-rime-deploy";
-        icon = "fcitx_rime_deploy";
         if (messageValue == "start") {
             message = _("Rime is under maintenance. It may take a few "
                         "seconds. Please wait until it is finished...");
+            icon = "info";
         } else if (messageValue == "success") {
             message = _("Rime is ready.");
+            icon = "success";
             if (!api_->is_maintenance_mode()) {
                 if (needRefreshAppOption_) {
                     api_->deploy_config_file("fcitx5.yaml", "config_version");
@@ -541,6 +565,7 @@ void RimeEngine::notify(RimeSessionId session, const std::string &messageType,
             needRefreshAppOption_ = false;
             message = _("Rime has encountered an error. "
                         "See log for details.");
+            icon = "error";
             blockMessage = true;
         }
     } else if (messageType == "option") {
