diff --git a/src/zhuyinengine.cpp b/src/zhuyinengine.cpp
index 7554e1d..d0e220c 100644
--- a/src/zhuyinengine.cpp
+++ b/src/zhuyinengine.cpp
@@ -553,11 +553,6 @@ void ZhuyinEngine::reloadConfig() {
 
     zhuyin_set_options(context_.get(), options);
 
-    instance_->inputContextManager().foreach([this](InputContext *ic) {
-        auto *state = ic->propertyFor(&factory_);
-        state->reset();
-        return true;
-    });
 
     zhuyin_load_phrase_library(context_.get(), USER_DICTIONARY);
 }
